### PR TITLE
feat(markdown): support text color in html style attribute

### PR DESCRIPTION
**Pull request by hannahu**
fix: https://github.com/Lundalogik/limepkg-email/issues/590

## Review:
- [ ] Commits are [atomic](https://seesparkbox.com/foundry/atomic_commits_with_git)
- [ ] Commits have the correct *type* for the changes made
- [ ] Commits with *breaking changes* are marked as such

### Browsers tested:
(Check any that applies, it's ok to leave boxes unchecked if testing something didn't seem relevant.)

Windows:
- [ ] Chrome
- [ ] Edge
- [ ] Firefox

Linux:
- [ ] Chrome
- [ ] Firefox

macOS:
- [ ] Chrome
- [ ] Firefox
- [ ] Safari

Mobile:
- [ ] Chrome on Android
- [ ] iOS

### CODE REVIEW

File: src/components/markdown/markdown-parser.ts [hunk 1]
```diff
@@ -35,7 +35,14 @@ export async function markdownToHTML(
         .use(remarkRehype, { allowDangerousHtml: true })
         .use(rehypeExternalLinks as any, { target: '_blank' })
         .use(rehypeRaw)
-        .use(rehypeSanitize)
+        .use(rehypeSanitize as any, {
+            attributes: {
+                '*': ['style'],
+            },
+            style: {
+                color: /^#([a-fA-F0-9]{6}|[a-fA-F0-9]{3})$/,
```
Comment (adrianschmidt) [inline src/components/markdown/markdown-parser.ts:43]: It seems like this isn't filtering anything 🙁

image

I'll look further into this tomorrow.

Comment (hannahu) [reply src/components/markdown/markdown-parser.ts:43]: Ahh, I didn't test it further. I blame chatgpt for the syntax 😄

Comment (adrianschmidt) [reply src/components/markdown/markdown-parser.ts]: ⚡

File: src/components/markdown/markdown-parser.ts [hunk 2]
```diff
@@ -35,7 +37,16 @@ export async function markdownToHTML(
...
+        .use(rehypeSanitize, {
+            attributes: {
+                '*': ['style'],
+            },
+        })
+        .use(() => {
+            return (tree: any) => {
+                visit(tree, 'element', sanitizeStyle);
+            };
+        })
```
Comment (adrianschmidt) [inline src/components/markdown/markdown-parser.ts:49]: Hang on, I'll add some comments here.

File: src/components/markdown/markdown-parser.ts [hunk 3]
```diff
@@ -35,7 +37,19 @@ export async function markdownToHTML(
...
         .use(rehypeRaw)
-        .use(rehypeSanitize)
+        .use(rehypeSanitize, {
+            // Allow the `style` attribute on all elements
+            attributes: {
+                '*': ['style'],
+            },
+        })
+        .use(() => {
+            return (tree: any) => {
```
Comment (LawrenceBorst) [inline src/components/markdown/markdown-parser.ts:47]: Not important but we could typehint this as a `Node` from `unist`.

**Review by LawrenceBorst — APPROVED** [review]
Really very good. Tested it out myself with the various css properties. Always manages to parse the style if parseable. This looks safe.

The tests are excellent and I can't think of any further security vulnerabilities.

Let's handle linebreaks at a later time.

File: src/components/markdown/markdown-parser.ts (hunk 3 above)
Comment (adrianschmidt) [reply src/components/markdown/markdown-parser.ts:47]: Fixed in https://github.com/Lundalogik/lime-elements/pull/2891

**Pull request merged**